assets/app/scripts/directives/truncate.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+angular.module('openshiftConsole')
+  // Truncates text to a length, adding a tooltip and an ellipsis if truncated.
+  // Different than `text-overflow: ellipsis` because it allows for multiline text.
+  .directive('truncateLongText', function() {
+    return {
+      restrict: 'E',
+      scope: {
+        content: '=',
+        limit: '=',
+        useWordBoundary: '='
+      },
+      // Keep everything on one line in the template to prevent unwanted linebreaks if `white-space` is set.
+      templateUrl: 'views/directives/_truncate.html',
+      link: function(scope, elem, attr) {
+        if (!scope.limit) {
+          return;
+        }
+
+        scope.$watch('content', function(content) {
+          if (!content || content.length <= scope.limit) {
+            scope.truncatedContent = null;
+            return;
+          }
+
+          scope.truncatedContent = content.substring(0, scope.limit);
+          if (scope.useWordBoundary) {
```

Comment:
I had that originally, but wasn't necessary. When I tried `use-word-boundary="false"` in my testing, Angular set `scope.useWordBoundary` to the boolean value `false`.